web/source/osk/embedded/subkeyDelegator.ts

```diff
@@ -0,0 +1,87 @@
+/// <reference path="../realizedGesture.interface.ts" />
+
+namespace com.keyman.osk.embedded {
+  // "Delegator", rather than "Popup", because KMW delegates display + selection
+  // of subkeys to the host app when in the embedded context.
+  export class SubkeyDelegator implements RealizedGesture {
+    private resolver: (keyEvent: text.KeyEvent) => void;
+    private readonly vkbd: VisualKeyboard;
+
+    public readonly baseKey: KeyElement;
+    public readonly promise: Promise<text.KeyEvent>;
+
+    private movedFromBaseKey: boolean = false;
+    private baseKeySelected: boolean = false;
+
+    constructor(vkbd: VisualKeyboard, e: KeyElement) {
+      this.vkbd = vkbd;
+
+      let _this = this;
+      this.promise = new Promise<text.KeyEvent>(function(resolve) {
+        _this.resolver = resolve;
+      });
+
+      this.baseKey = e;
+    }
+
+    public resolve(keyCoreID: string) {
+      if(this.resolver) {
+        let keyEvent: text.KeyEvent = null;
+
+        if(keyCoreID == null && this.baseKeySelected) {
+          // Handle selection of base key underneath the subkey array.
+          keyEvent = this.vkbd.keyEventFromSpec(this.baseKey.key.spec as keyboards.ActiveKey, null);
+          this.baseKey.key.highlight(false);
+        } else {
+          // This is set with the base key of our current subkey elsewhere within the engine.
+          var baseKey: OSKKeySpec = this.baseKey.key.spec;
```

Comment:
```suggestion
          let baseKey: OSKKeySpec = this.baseKey.key.spec;
```

Query, we use `let` in places where we could use `const`. What do you think about moving towards `const` for vars that are never changed?

Reply:
That'd be a _lot_ easier to do if I weren't also maintaining our Swift codebase... where Swift's `let` _**is**_ the TS `const`.  Which is why I sometimes slip `var` in, as that's basically `TS`'s `let`.

Reply:
Yeah, well, welcome to cross-language development I guess? 🤣

Reply:
Yeah.  It's _also_ fun because you can't use `const` for class fields - you want `readonly`..  I ran up against that so many times I legitimately forget that `const` can be used for local variables in TS.